### DISCUSSION TITLE
i18 needs support for translating urls too

### DISCUSSION THREAD

**Discussion by sebastian-nowak** [vercel/next.js#21152] [Ideas]
**Describe the feature you'd like to request**

Unless I'm missing something, currently it's possible to create prefixed versions of a page, but the main part of the url stays the same across languages. The same thing happens when using different domains for different locales.

Currently I can only get URLs like:
* domain.com/about
* domain.pl/about
* domain.de/about

We really need a way to translate the URLs too, so we can get these URLs:
* domain.com/about
* domain.pl/o-nas
* domain.de/yeah-i-dont-speak-german-but-you-get-the-idea

**Describe the solution you'd like**

I guess ideally a file in pages/* could export a method that returns a path for each locale. Path for the defaultLocale would be derived from the filesystem path in the same way it works now. Or not, maybe it should be explicitly defined too.

**Describe alternatives you've considered**

🤷‍♂️

**Comment by dmudro** [comment]
Also in this summary: https://github.com/vercel/next.js/discussions/20240
